orcestra/postprocess/level1.py

```diff
@@ -48,50 +49,78 @@ def _state_filter_radar(ds):
     return ds.where(ds.grst == config["valid_radar_state"])
 
 
-def _roll_filter(ds, ds_bahamas):
+def _roll_filter(ds, roll):
     """Filter any dataset by plane roll angle.
 
     Parameters
     ----------
     ds : xr.Dataset
         Level0 dataset.
-    ds_bahamas : xr.Dataset
-        Level0 BAHAMAS dataset.
+    roll: xr.DataArray
+        Flight roll angle in degrees.
 
     Returns
     -------
     xr.Dataset
         Dataset filtered by plane roll angle.
     """
 
-    ds_bahamas_subsampled = ds_bahamas.sel(
+    roll_subsampled = roll.sel(
         time=ds.time, method="nearest"
     ).assign_coords(time=ds.time)
 
-    return ds.where(np.abs(ds_bahamas_subsampled.IRS_PHI) < config["roll_threshold"])
+    return ds.where(np.abs(roll_subsampled) < config["roll_threshold"])
 
 
-def _altitude_filter(ds, ds_bahamas):
+def _altitude_filter(ds, height):
     """Filter any dataset by plane altitude.
 
     Parameters
     ----------
     ds : xr.Dataset
         Level0 dataset.
-    ds_bahamas : xr.Dataset
-        Level0 BAHAMAS dataset.
+    height: xr.DataArray
+        Flight altitude in m.
 
     Returns
     -------
     xr.Dataset
         Dataset filtered by plane altitude.
     """
 
-    ds_bahamas_subsampled = ds_bahamas.sel(
+    height_subsampled = height.sel(
         time=ds.time, method="nearest"
     ).assign_coords(time=ds.time)
 
-    return ds.where(ds_bahamas_subsampled.IRS_ALT > config["altitude_threshold"])
+    return ds.where(height_subsampled > config["altitude_threshold"])
+
+def _trim_dataset(ds, dim="time"):
+    """
+    Trim the dataset by removing data at the beginning and end until the first and last occurrence of valid data.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The input dataset.
+    dim : str
+        The dimension along which to trim the dataset. Default is "time".
+
+    Returns
+    -------
+    xr.Dataset
+        The trimmed dataset.
+    """
+    # Drop NaNs along the specified dimension
+    valid_data = ds.dropna(dim=dim, how="all")
```

Comment:
Isn't `dropna` returning the trimmed dataset already? Or do you want to explicitly keep NaNs in the middle of the time series?